src/middleware/api-documentation.ts

```diff
@@ -0,0 +1,18 @@
+import Router from '@koa/router';
+import Link from 'http-link-header';
+import { Context, Middleware, Next } from 'koa';
+import { hydra } from 'rdf-namespaces';
+import Routes from '../routes';
+
+export default (router: Router): Middleware => (
+  async ({ response }: Context, next: Next): Promise<void> => {
+    await next();
+
+    const link = new Link().set({
+      rel: hydra.apiDocumentation,
+      uri: router.url(Routes.ApiDocumentation, {}),
+    });
+
+    response.append('Link', link.toString());
```

Comment:
Try [set-link](https://www.npmjs.com/package/set-link)

```suggestion
    response.setLink(router.url(Routes.ApiDocumentation, {}), hydra.apiDocumentation)
```

Reply:
Had a look, have to let TypeScript know about the new method so gone with yet another library in https://github.com/libero/article-store/pull/14/commits/d23e4347f483c3ffe76b0c34bc6ba6947b53033e.

Reply:
How does express behave when you call `append('Link')` multiple times? `set-link` nicely merges multiple calls.

If typings are an issue, you may enjoy DefinitelyTyped/DefinitelyTyped#40374

Reply:
Not sure, this is Koa. 😅

(Not sure how Koa does it either. Probably the same.)

Reply:
Would prefer a syntax like `set-link` though. Maybe can refactor later.